src/puppetlabs/trapperkeeper/authorization/ring_middleware.clj

```diff
@@ -209,11 +209,14 @@
    allow-header-cert-info :- schema/Bool
    oid-map :- acl/OIDMap]
   (if-let [cert (request->cert request allow-header-cert-info)]
-    (let [extensions (ssl-utils/get-extensions cert)
+    (let [oid-map (merge acl/default-oid-map oid-map)
+          extensions (ssl-utils/get-extensions cert)
           translate-oids (fn [out extension]
                            (let [oid-key (get oid-map (:oid extension)
                                               (keyword (:oid extension)))
-                                 value (str (:value extension))]
+                                 value (if (= :subjectAlternativeName oid-key)
+                                         (ssl-utils/get-subject-dns-alt-names cert)
```

Comment:
Makes perfect sense, and I was wondering if I should be doing this exact thing. Thanks for bringing it up.

This will take me a bit of time to add since it means special case parsing on the config side as well as the request side. I'll try to have the PR updated by eod monday.